roqoqo/src/noise_models/continuous_decoherence.rs

```diff
@@ -0,0 +1,344 @@
+        for qubit in qubits {
+            // This can never fail here
+            self.lindblad_noise
+                .add_operator_product(
+                    (
+                        // In the qc convention in a single qubit state vector v[0] corresponds to state |0>
+                        // and v[1] corresponds to state |1>
+                        // The plus-minus plus operator brings state |1> to state |0>
+                        // Therefor the plus operator corresponds to damping
+                        PlusMinusProduct::new().plus(*qubit),
+                        PlusMinusProduct::new().plus(*qubit),
+                    ),
+                    rate.into(),
+                )
+                .expect("Internal struqture bug.");
+        }
+        self
+    }
+    /// Convenience function to add rate of excitation from 0 to 1 state to several qubits
+    ///
+    /// # Arguments
+    ///
+    /// * `qubits` - The qubits to add the rate of excitation to.
+    /// * `rate` - The rate of excitation.
+    ///
+    /// # Returns
+    ///
+    /// * `self` - The ContinuousDecoherenceModel with the rate of excitation added.
+    pub fn add_excitation_rate(mut self, qubits: &[usize], rate: f64) -> Self {
+        for qubit in qubits {
+            // This can never fail here
+            self.lindblad_noise
+                .add_operator_product(
+                    (
+                        PlusMinusProduct::new().minus(*qubit),
+                        PlusMinusProduct::new().minus(*qubit),
+                    ),
+                    rate.into(),
+                )
+                .expect("Internal struqture bug.");
+        }
+        self
+    }
+    /// Convenience function to add damping to several qubits
+    ///
+    /// # Arguments
+    ///
+    /// * `qubits` - The qubits to add dephasing to.
+    /// * `rate` - The dephasing rate.
+    ///
+    /// # Returns
+    ///
+    /// * `self` - The ContinuousDecoherenceModel with the dephasing added.
+    pub fn add_dephasing_rate(mut self, qubits: &[usize], rate: f64) -> Self {
+        for qubit in qubits {
+            // This can never fail here
+            self.lindblad_noise
+                .add_operator_product(
+                    (
+                        PlusMinusProduct::new().z(*qubit),
+                        PlusMinusProduct::new().z(*qubit),
+                    ),
+                    rate.into(),
+                )
+                .expect("Internal struqture bug.");
+        }
+        self
+    }
+    /// Convenience function to add depolarising to several qubits
+    ///
+    /// # Arguments
+    ///
+    /// * `qubits` - The qubits to add depolarising to.
+    /// * `rate` - The rate of depolarising.
+    ///
+    /// # Returns
+    ///
+    /// * `self` - The ContinuousDecoherenceModel with the rate of depolarising added.
+    pub fn add_depolarising_rate(mut self, qubits: &[usize], rate: f64) -> Self {
```

Comment:
This function was not mentioned in the documentation above the struct, while the other three functions were - should it be added?